datajunction-server/datajunction_server/api/main.py

```diff
@@ -37,84 +39,119 @@
+app.include_router(data.router)
+app.include_router(health.router)
+app.include_router(history.router)
+app.include_router(cubes.router)
+app.include_router(tags.router)
+app.include_router(attributes.router)
+app.include_router(sql.router)
+app.include_router(client.router)
+app.include_router(dimensions.router)
+
+
+@app.exception_handler(DJException)
+async def dj_exception_handler(  # pylint: disable=unused-argument
+    request: Request,
+    exc: DJException,
+) -> JSONResponse:
     """
-    Get the DJ FastAPI app and optionally inject an OpenTelemetry tracer provider
+    Capture errors and return JSON.
     """
-    settings = get_settings()
-    config.fileConfig(
-        path.join(path.dirname(path.abspath(__file__)), "logging.conf"),
-        disable_existing_loggers=False,
-    )
-    application = FastAPI(
-        title=settings.name,
-        description=settings.description,
-        version=__version__,
-        license_info={
-            "name": "MIT License",
-            "url": "https://mit-license.org/",
-        },
-        dependencies=[Depends(default_attribute_types)],
-    )
-    application.add_middleware(
-        CORSMiddleware,
-        allow_origins=settings.cors_origin_whitelist,
-        allow_credentials=True,
-        allow_methods=["*"],
-        allow_headers=["*"],
+    _logger.exception(exc)
+    return JSONResponse(
+        status_code=exc.http_status_code,
+        content=exc.to_dict(),
+        headers={"X-DJ-Error": "true", "X-DBAPI-Exception": exc.dbapi_exception},
     )
 
-    application.include_router(catalogs.router)
-    application.include_router(engines.router)
-    application.include_router(metrics.router)
-    application.include_router(djsql.router)
-    application.include_router(nodes.router)
-    application.include_router(namespaces.router)
-    application.include_router(materializations.router)
-    application.include_router(data.router)
-    application.include_router(health.router)
-    application.include_router(history.router)
-    application.include_router(cubes.router)
-    application.include_router(tags.router)
-    application.include_router(attributes.router)
-    application.include_router(sql.router)
-    application.include_router(client.router)
-    application.include_router(dimensions.router)
-
-    @application.exception_handler(DJException)
-    async def dj_exception_handler(  # pylint: disable=unused-argument
-        request: Request,
-        exc: DJException,
-    ) -> JSONResponse:
-        """
-        Capture errors and return JSON.
-        """
-        _logger.exception(exc)
-        return JSONResponse(
-            status_code=exc.http_status_code,
-            content=exc.to_dict(),
-            headers={"X-DJ-Error": "true", "X-DBAPI-Exception": exc.dbapi_exception},
-        )
-
-    FastAPIInstrumentor.instrument_app(application, tracer_provider=tracer_provider)
-
-    return application
-
-
-app = get_dj_app()
```

Comment:
Any reason why this is changed to creating the app directly rather than via this function call?

Reply:
TLDR; the value of clarity in this file is starting to feel worth more than the value of having tracing in the demo environment.

I initially added the function call to allow injecting a custom open telemetry tracer configuration, but this `main.py` is only really used for the demo. So having the otel stuff here didn't really seem informative or useful since our recommendation for someone wanting to instrument otel would probably be to mount all the routers to their own app instance and follow any standard instructions out there for otel instrumentation in fastapi.